scripts/index.js

```diff
@@ -114,27 +118,32 @@ function renderMovieResults() {
             </div>
             <div class="movie-details">    
                 <div class="movie-title">      
-                        <h2> ${data.Title}</h2>
-                        <p> ${data.Ratings[0] === undefined ? "Rating unavailable" : data.Ratings[0]?.Value.split("/")[0]} <i class="fa-solid fa-star"></i></p>
+                    <h2> ${data.Title}</h2>
+                    <p>
+                        ${data.Ratings[0] === undefined ? "Rating unavailable" : data.Ratings[0]?.Value.split("/")[0]}
+                        <i class="fa-solid fa-star"></i>
+                    </p>
                 </div>
-                        <div class="small-detail">
-                            <p> ${data.Runtime === "N/A" ? "Runtime unavailable" : data.Runtime} </p>
-                            <p> ${data.Genre === "N/A" ? "Genre unavailable" : data.Genre} </p>
-                            
-                            <button data-add-button data-add-id="${data.imdbID}" class="watchlist-btn"> ${btnInnerHtml} </button>
-                        </div>
-                        <div class="about-detail"> 
-                            <p>${data.Plot === "N/A" ? "No description available" : data.Plot}</p> 
-                        </div> 
+                <div class="small-detail">
+                    <p> ${data.Runtime === "N/A" ? "Runtime unavailable" : data.Runtime} </p>
+                    <p> ${data.Genre === "N/A" ? "Genre unavailable" : data.Genre} </p>
+                    
+                    <button data-add-button data-add-id="${data.imdbID}" class="watchlist-btn"> ${btnInnerHtml} </button>
                 </div>
-            </div>    
-                `
+                <div class="about-detail"> 
+                    <p>${data.Plot === "N/A" ? "No description available" : data.Plot}</p> 
+                </div> 
+            </div>
+        </div>    
+        `
     })
+
     movieResultsElement.innerHTML = htmlString
+
     document.querySelectorAll("[data-add-button]").forEach(button => {
         button.addEventListener('click', () => {
-            saveToWatchlist(button.dataset.addId)
+            saveToWatchlist(button.dataset.addId, movieResults)
```

Comment:
As `saveToWatchList` needs to add the whole 'movie' object to the watchlist, rather than just the movie's id, we need to pass `movieResults` to the function.